app/platform/hw_timer.c

```diff
@@ -0,0 +1,182 @@
+/******************************************************************************
+* Copyright 2013-2014 Espressif Systems (Wuxi)
+*
+* FileName: hw_timer.c
+*
+* Description: hw_timer driver
+*
+* Modification history:
+*     2014/5/1, v1.0 create this file.
+*     
+* Adapted for NodeMCU 2016
+* 
+* The owner parameter should be a unique value per module using this API
+* It could be a pointer to a bit of data or code 
+* e.g.   #define OWNER    ((os_param_t) module_init)   
+* where module_init is a function. For builtin modules, it might be 
+* a small numeric value that is known not to clash.
+*******************************************************************************/
+#include "ets_sys.h"
+#include "os_type.h"
+#include "osapi.h"
+
+#include "hw_timer.h"
+
+#define FRC1_ENABLE_TIMER  BIT7
+#define FRC1_AUTO_LOAD  BIT6
+
+//TIMER PREDIVIDED MODE
+typedef enum {
+    DIVIDED_BY_1 = 0,		//timer clock
+    DIVIDED_BY_16 = 4,	//divided by 16
+    DIVIDED_BY_256 = 8,	//divided by 256
+} TIMER_PREDIVIDED_MODE;
+
+typedef enum {			//timer interrupt mode
+    TM_LEVEL_INT = 1,	// level interrupt
+    TM_EDGE_INT   = 0,	//edge interrupt
+} TIMER_INT_MODE;
+
+static os_param_t the_owner;
```

Comment:
There is implicit initialization of statics to zero at start time (this is the .bss section). If you initialize a static, then the initialisation value goes into the .data segment and this needs to be copied into RAM at start time. 

In my early days of programming (in the '70s) you learned to squeeze every byte out of the code -- and this habit of mine with regard to not explicitly initialising static variables to zero dates from when I started to use C in the '80s.